kuma/static/js/components/payments/focus-visible.js

```diff
@@ -0,0 +1,25 @@
+(function() {
+    'use strict';
+    var contributeForm = document.getElementById('contribute-form');
+
+    // When a keyup event is triggered on the contribute form,
+    contributeForm.addEventListener('keyup', function(event) {
+        var currentTarget = event.target;
+        var targetType = currentTarget.type;
+
+        //  and the target field type is one of `text`, `email`, or `number
+        if (
+            targetType === 'text' ||
+            targetType === 'email' ||
+            targetType === 'number'
+        ) {
+            // add the `focus-visible` class.
+            currentTarget.classList.add('focus-visible');
+            // listen for when the field loses focus,
+            currentTarget.addEventListener('focusout', function() {
```

Comment:
I wish the DOM had something like the jQuery 'once' event handler registration method...

This focusout handler never gets removed, but gets added multiple times, so if the user tabs around through the form a lot, I think you're going to end up calling classList.remove() multiple times.

Would it work to just have a focusout handler on contributeForm in parallel to the keyup handler?

Reply:
I was a bit concerned about that as well. I believe I have found a way to address it.